msal/src/main/java/com/microsoft/identity/client/AuthenticationResult.java

```diff
@@ -39,30 +39,28 @@
 public final class AuthenticationResult implements IAuthenticationResult {
     //Fields for Legacy Cache
     private final String mTenantId;
-    private final String mRawIdToken;
     private final String mUniqueId;
 
     private final AccessTokenRecord mAccessToken;
-    private final IAccount mAccount;
-
-    public AuthenticationResult(@NonNull final ICacheRecord cacheRecord) {
-        mAccessToken = cacheRecord.getAccessToken();
-        mTenantId = cacheRecord.getAccount().getRealm();
-        mUniqueId = cacheRecord.getAccount().getHomeAccountId();
-        mRawIdToken = cacheRecord.getIdToken().getSecret();
-        mAccount = AccountAdapter.adapt(cacheRecord.getAccount());
+    private final com.microsoft.identity.client.tenantprofile.IAccount mAccount;
+
+    public AuthenticationResult(@NonNull final List<ICacheRecord> cacheRecords) {
+        final ICacheRecord mostRecentlyAuthorized = cacheRecords.get(0);
+        mAccessToken = mostRecentlyAuthorized.getAccessToken();
+        mTenantId = mostRecentlyAuthorized.getAccount().getRealm();
+        mUniqueId = mostRecentlyAuthorized.getAccount().getHomeAccountId();
+        mAccount = com.microsoft.identity.client.tenantprofile.AccountAdapter.adapt(cacheRecords).get(0);
     }
 
-    public AuthenticationResult(@NonNull AccessTokenRecord accessToken,
-                                @Nullable String rawIdToken,
-                                @NonNull IAccountRecord accountRecord) {
-        mAccessToken = accessToken;
-        mTenantId = accessToken.getRealm();
-        mUniqueId = accessToken.getHomeAccountId();
-        mRawIdToken = rawIdToken;
-        mAccount = AccountAdapter.adapt(accountRecord);
-
-    }
+//    public AuthenticationResult(@NonNull AccessTokenRecord accessToken,
```

Comment:
Are you planning to delete or rewrite this method?

Reply:
Resolved in 71aa5d5760998128a65685013e351387d5338ac3